src/mainboard/clevo/tgl-u/ramstage.c

```diff
@@ -107,11 +116,46 @@ static void set_fan_curve(void)
 	}
 }
 
+static void set_camera_enablement(void)
+{
+    struct smfi_cmd_camera_enablement_set cmd;
+    bool enabled = CAMERA_ENABLEMENT_DEFAULT;
+
+#if CONFIG(DRIVERS_EFI_VARIABLE_STORE)
+    struct region_device rdev;
+    enum cb_err ret;
+    uint32_t size;
+
+    const EFI_GUID dasharo_system_features_guid = {
+		0xd15b327e, 0xff2d, 0x4fc1, { 0xab, 0xf6, 0xc1, 0x2b, 0xd0, 0x8c, 0x13, 0x59 }
+	};
+
+    if (smmstore_lookup_region(&rdev))
+		goto efi_err;
+
+    size = sizeof(enabled);
+    ret = efi_fv_get_option(&rdev, &dasharo_system_features_guid, "EnableCamera", &enabled, &size);
+
+    if (ret != CB_SUCCESS) {
+		printk(BIOS_DEBUG, "camera: failed to read camera enablement status from EFI vars, using board default\n");
+		goto efi_err;
+    }
+
+efi_err:
+#endif
+
+    cmd.enable = enabled;
+    system76_ec_smfi_cmd(CMD_CAMERA_ENABLEMENT_SET, sizeof(cmd) / sizeof(uint8_t), (uint8_t *)&cmd);
```

Comment:
```suggestion
    system76_ec_smfi_cmd(CMD_CAMERA_ENABLEMENT_SET, sizeof(enabled), (uint8_t *)&enabled);
```

This should be enough

Reply:
Then we can also get rid off `struct smfi_cmd_camera_enablement_set`